pymeasure/instruments/racal/racal1992.py

```diff
@@ -0,0 +1,363 @@
+            'trigger_level' : None,                                     # noqa
+            },
+        'B' : {                                                         # noqa
+            'coupling'      : { 'AC'        : 'BAC',  'DC'     : 'BDC' },    # noqa
+            'attenuation'   : { 'X1'        : 'BAD',  'X10'    : 'BAE' },    # noqa
+            'trigger'       : { 'auto'      : 'BAU',  'manual' : 'BMN' },    # noqa
+            'impedance'     : { '50'        : 'BLI',  '1M'     : 'BHI' },    # noqa
+            'slope'         : { 'pos'       : 'BPS',  'neg'    : 'BNS' },    # noqa
+            'input_select'  : { 'separate'  : 'BCS',  'common' : 'BCC' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+    }
+
+    operating_modes = {
+        'self_check'      : 'CK',    # noqa
+        'frequency_a'     : 'FA',    # noqa
+        'period_a'        : 'PA',    # noqa
+        'phase_a_rel_b'   : 'PH',    # noqa
+        'ratio_a_to_b'    : 'RA',    # noqa
+        'ratio_c_to_b'    : 'RC',    # noqa
+        'interval_a_to_b' : 'TI',    # noqa
+        'total_a_by_b'    : 'TA',    # noqa
+        'frequency_c'     : 'FC',    # noqa
+    }
+
+    def decode(v, allowed_types=None):
+        """Decode received message.
+
+        All values returned follow the same format: 2 letters to indicate
+        the type of the value returned, followed by a floating point number
+        (which could be an integer, of course.)
+        This here, for example, is math constant Z: MZ+001.00000000E+00
+        """
+        if len(v) != 19:
+            raise ReturnValueError("Length of instrument response must always be 19 characters")
+
+        val_type = v[0:2]
+        val = float(v[2:19])
+
+        if allowed_types and val_type not in allowed_types:
+            raise ValueError(f"Unexpected value type returned: '{val_type}'")
+
+        if val_type in Racal1992.int_types:
+            return int(val)
+        elif val_type in Racal1992.float_types:
+            return val
+        else:
+            raise ValueError("Unsupported return type")
+
+    operating_mode = Instrument.setting(
+            "%s",
+            """ Set operating mode.
+
+            Permitted modes are:
+                'self_check',
+                'frequency_a',
+                'period_a',
+                'phase_a_rel_b',
+                'ratio_a_to_b',
+                'ratio_c_to_b',
+                'interval_a_to_b',
+                'total_a_by_b',
+                'frequency_c'
+            """,
+            set_process=(lambda mode: Racal1992.operating_modes[mode]),
+        )
+
+    resolution = Instrument.control(
+            "RRS",
+            "SRS %d",
+            """ Control the resolution of the counter with an integer from 3 to 10 that
+            specifies the number of significant digits. """,
+            get_process=(lambda v: Racal1992.decode(v, "RS"))
```

Comment:
```suggestion
            get_process=Racal1992.decode("RS")
```
I think you could use this syntax, if you change `decode` in a function factory based on the original `decode` function, something like:
```
def decode(allowed_types=None):
   return partial(former_decode(allowed_types=allowed_types)
```
where `former_decode` is your original decode static method.
You need also to include partial as follow:
`from functools import partial`

Anyway, not a strong suggestion, I would leave to your decision whether you like it and it's worth the effort.

Reply:
Same as the previous comment. The current code works, the suggest change does not. It's an immaterial change.